docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,346 @@
+            var embedInteropTypes = (embedInteropTypesAndKind & 0b10) == 0b10;
+            var kind = (embedInteropTypesAndKind & 0b1) == 0b1
+                ? MetadataImageKind.Assembly
+                : MetadataImageKind.Module;
+
+            var timestamp = blobReader.ReadInt32();
+            var imageSize = blobReader.ReadInt32();
+            var mvid = blobReader.ReadGuid();
+
+            Console.WriteLine(name);
+            Console.WriteLine($"Extern Aliases: \"{externAliases}\"");
+            Console.WriteLine($"Embed Interop Types: {embedInteropTypes}");
+            Console.WriteLine($"Metadata Image Kind: {kind}");
+            Console.WriteLine($"Timestamp: {timestamp}");
+            Console.WriteLine($"Image Size: {imageSize}");
+            Console.WriteLine($"MVID: {mvid}");
+            Console.WriteLine();
+        }
+    }
+}
+```
+
+#### Compiler Flag Key Value Pairs
+
+The remaining values will be stored as key value pairs in the pdb. The storage format will be UTF8 encoded key value pairs that are null terminated. Order is not guaranteed. Any values left out can be assumed to be the default for the type. Keys may be different for Visual Basic and CSharp. They are serialized to reflect the command line arguments representing the same values
+
+Example: 
+
+`compilerversion\01.2.3-example-sha\0sourceencoding\0utf-8\0checked\01\0unsafe\01\0langversion\0latest\0nullable\0Enable` 
+
+##### Options For CSharp
+
+See [compiler options](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/listed-alphabetically) documentation
+
+* "[appconfig](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/appconfig-compiler-option)"
+    - Since appconfig is a pointer to a file not embedded in the PDB or PE, we instead extract the information that will directly impact the compilation. We store this as a flag called `portability-policy` with a numeric value from \[0-3\]. This value directly correlates to reconstructing the [AssemblyPortabilityPolicy](https://github.com/dotnet/roslyn/blob/bdb3ece74c85892709f5e42ae7d67248999ecc3b/src/Compilers/Core/Portable/Desktop/AssemblyPortabilityPolicy.cs).
+        - 0 -> SuppressSilverlightPlatformAssembliesPortability = false, SuppressSilverlightLibraryAssembliesPortability = false
+        - 1 -> SuppressSilverlightPlatformAssembliesPortability = true, SuppressSilverlightLibraryAssembliesPortability = false
+        - 2 -> SuppressSilverlightPlatformAssembliesPortability = false, SuppressSilverlightLibraryAssembliesPortability = true
+        - 3 -> SuppressSilverlightPlatformAssembliesPortability = true, SuppressSilverlightLibraryAssembliesPortability = true
+* "[checked](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/checked-compiler-option)"
+* "[codepage](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/codepage-compiler-option)"
+* "compilerversion"
```

Comment:
I thought we discussed that runtime version wouldn't be needed. I'm going to reread https://github.com/dotnet/roslyn/pull/44373#discussion_r431280947

Reply:
I don't have a strong opinion on multiple words with hyphen, I just did it as is because that's how cli works. Happy to go either way if you feel strongly about it.